app/src/main/java/org/sil/storyproducer/activities/BaseActivity.kt

```diff
@@ -81,10 +81,22 @@ open class BaseActivity : AppCompatActivity(), BaseActivityView {
         startActivity(Intent(this, MainActivity::class.java))
         finish()
     }
-
-    override fun showRegistration() {
+    // DKH - 05/12/2021
+    // Issue #573: SP will hang/crash when submitting registration
+    // showRegistration Argument allows calling function in current Activity to terminate the
+    // current Activity and then show the Registration screen via the RegistrationActivity
+    // Any MainActivity function should set executeFinishActivity to false.  MainActivity should
+    // always exist.
+    // Activities like LearnActivity will exit by using the default value of true
+    // After the RegistrationActivity completes, control is returned to the MainActivity
+    // where the list of story templates are displayed
+    override fun showRegistration(executeFinishActivity: Boolean) {
         startActivity(Intent(this, RegistrationActivity::class.java))
```

Comment:
Help me understand this. Does this startActivity() initiate a new activity, superceding the current activity (either mainActivity or a phaseActivity (eg LearnActivity)?
Would it make sense for the phaseActivity to have an override of showRegistration (or something else) that finishes the phaseActivity and thus reverts the app to mainActivity as it calls this baseActivity showRegistration() to do the registrationActivity?

Reply:
Can the same control level that established the phaseActivity be used to finish that phaseActivity and thus revert to the MainActivity, which can then show the Registration?